### DISCUSSION TITLE
Reusing a list for several targets without having to re-run those targets?

### DISCUSSION THREAD

**Discussion by nmhare** [ropensci/targets#472] [Ideas]
Hello,

I am using Targets to create several different figures in the same file, figures.R. At the same time, the color palette for these figures is contained in a different file, aesthetics.R. This way, the specifics of a figure's design (what type of graph, what data to include, etc.) is separate from the artistic choices we make.

Inside aesthetics.R, a list contains information for each figure, and figures.R references this list. If I change any information within the list, Targets will notice a change and re-run every figure referencing this list, even if it hasn't influenced that figure. I'm wondering if Targets can manage to recognize that only one element in the `visuals` list changed, and not re-run the target for both figure 1 and figure 2.

Here is a short example. If, in aesthetics.R, I change figure 1's color mapping, Targets will re-run figure 2's target. I only want figure 1 to rerun:

```
#_targets.R
source("src/aesthetics.R")
...
tar_options_set(...)
...

list(
    tar_target(data, create_data("path/to/file")),
    tar_target(figure_1, create_figure_1(data)),
    tar_target(figure_2, create_figure_2(data))
)
```

```
#figures.R
create_figure_1 <- function(data) {
    ...
}
create_figure_2 <- function(data) {
    ...
}
```

```
#aesthetics.R
visuals <- list(
   figure_1 <- list(
       name  = "Figure 1 Name",
       values = c("color_factor_1" = "blue", "color_factor_2" = "orange") #changing these lists will make figure 2's target run again
    ), 
  figure_2 <- list(
      name  = "Figure 2 Name", # the same is true in the opposite direction - changing these reruns figure 1
      values = c("color_factor_1" = "red" , "color_factor_2" = "green") 
   )
)
```

I've thought of a workaround - instead of having a two-dimensional list, `visuals`, I just create several lists, one for each figure. However, I would prefer to keep the `visuals` list as one centralized object.

**Comment by wlandau** [comment]
Yeah, unfortunately you would need to keep separate aesthetics lists in order to avoid invalidating all the figures. You could write separate objects or embed the aesthetics in the functions that need them, or pass them as arguments during the targets' commands.
